FuelSDK-CSharp/ETClient.cs

```diff
@@ -110,9 +110,42 @@ public ETClient(NameValueCollection parameters = null, RefreshState refreshState
                 {
                     configSection.Scope = parameters["scope"];
                 }
+                if (parameters.AllKeys.Contains("applicationType"))
+                {
+                    configSection.ApplicationType = parameters["applicationType"];
+                }
+                if (parameters.AllKeys.Contains("authorizationCode"))
+                {
+                    configSection.AuthorizationCode = parameters["authorizationCode"];
+                }
+                 if (parameters.AllKeys.Contains("redirectURI"))
+                {
+                    configSection.RedirectURI = parameters["redirectURI"];
+                }
+            }
+
+            configSection.ApplicationType = string.IsNullOrEmpty(configSection.ApplicationType) ? "server" : configSection.ApplicationType;
+
+            if (configSection.ApplicationType.Equals("public") || configSection.ApplicationType.Equals("web"))
+            {
+                if (string.IsNullOrEmpty(configSection.AuthorizationCode) || string.IsNullOrEmpty(configSection.RedirectURI))
```

Comment:
Since we throw an exception here if AuthorizationCode or RedirectURI are NullOrEmpty, then, in the CreatePayload method, in the `else if (!config.ApplicationType.Equals("server"))` branch, we do not need the validations. This should be enoguh:

            else if (!config.ApplicationType.Equals("server"))
            {
                payload.grant_type = "authorization_code";
                payload.code = config.AuthorizationCode;
                payload.redirect_uri = config.RedirectURI;
            }